### PR TITLE
Improve IoT project reconciliation; 

### PR DESCRIPTION
**Pull request by dejanb**
### Type of change

- Enhancement / new feature

### Description

Update status only if there are changes

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by dejanb** [comment]
@enmasse-ci run tests profile=iot

**Comment by dejanb** [comment]
With this change I don't see events being created. But now there's an error like

```2020-01-17T14:27:40.543Z WARN  [DefaultChannelPipeline] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.channel.unix.Errors$NativeIoException: syscall:read(..) failed: Connection reset by peer
	at io.netty.channel.unix.FileDescriptor.readAddress(..)(Unknown Source)```

File: pkg/controller/iotproject/iotproject_controller.go [hunk 1]
```diff
@@ -261,9 +263,12 @@ func (r *ReconcileIoTProject) Reconcile(request reconcile.Request) (reconcile.Re
 		rc.Process(func() (result reconcile.Result, e error) {
 			return r.reconcileManaged(ctx, &request, project)
 		})
-		rc.Process(func() (result reconcile.Result, e error) {
-			return r.updateProjectStatus(ctx, project, rc.Error())
-		})
+		if !reflect.DeepEqual(project, original) {
+			reqLogger.Info("Project changed, updating status")
+			rc.Process(func() (result reconcile.Result, e error) {
```
Comment (ctron) [inline pkg/controller/iotproject/iotproject_controller.go]: Shouldn't the change be made on:

~~~go
	// update status
	err := r.client.Status().Update(ctx, newProject)
~~~

There are multiple paths leading to a call to `updateProjectStatus` and now I don't think we would properly update the status any longer.

**Comment by lulf** [comment]
@dejanb FWIW, a similar change might be needed for IoTConfig controller, as it has the same symptoms at least.

File: pkg/controller/iotproject/iotproject_controller.go (hunk 1 above)
Comment (dejanb) [reply pkg/controller/iotproject/iotproject_controller.go]: @ctron yeah ... This is the first thing that moved the needle in the local testing ... I want to see how it affects tests ... And will see how to better refactor all this if effective

**Comment by dejanb** [comment]
@lulf sure thing. Will keep looking

**Comment by dejanb** [comment]
@enmasse-ci run tests profile=iot

**Comment by dejanb** [comment]
@enmasse-ci run tests profile=iot

**Comment by dejanb** [comment]
@enmasse-ci run tests profile=iot

**Pull request merged**